moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -769,14 +715,20 @@ void RobotState::updateLinkTransformsInternal(const JointModel* start)
     }
     else  // is the origin / root / 'model frame'
     {
-      if (link->jointOriginTransformIsIdentity())
+      const JointModel* root_joint = link->getParentJointModel();
+      if (root_joint->getVariableCount() == 0)
       {
-        global_link_transforms_[idx_link] = getJointTransform(link->getParentJointModel());
+        // The root joint doesn't have any variables: avoid calling getJointTransform() on it.
+        global_link_transforms_[idx_link] = Eigen::Isometry3d::Identity();
```

Comment:
Calling `getJointTransform()` (or actually accessing the joint position of a fixed joint) just became a problem with this PR. Before, the (invalid) address of the position entry was passed but actually not accessed!

Reply:
Yes, the pointer API _can_ be misused, but I think it wasn't in the existing code base. However, this "risk" came for improved performance. I think the better performance in caching and memory allocation gets more pronounced if you are running short of memory, which nowadays is less than a problem than in the past...
In any case, we are balancing security for performance. With your PR you decided to weigh the former more.
Please also note that changing basic data structures also introduces the risk of new bugs - for example the missing initialization of accelerations in your PR.